xcm/pallet-xcm/src/lib.rs

```diff
@@ -685,12 +685,6 @@ pub mod pallet {
 			}
 			weight_used
 		}
-		fn on_runtime_upgrade() -> Weight {
```

Comment:
It's hard to check for all of them. Every migration should be tested with try-runtime which will yell at you if there is a storage mismatch for any pallet

Reply:
Is there a list of runtimes we can check off before merging this?

Reply:
I initially did that in this PR, but after reviews decided to only apply it to the chain I was upgrading (contracts-rococo)
https://github.com/paritytech/cumulus/pull/2570, (see also other earlier commit where I apply it to all runtimes https://github.com/paritytech/cumulus/pull/2570/commits/d8905708e182ea8b1b094980a806ef0f08b2aa33)

The reasoning is that adding a migration that have already been applied, adds some unnecessary bloat to the wasm runtime, so the best practice should be to test with try-runtime and see what needs to be included.

Reply:
following up, here since @vstam1  is trying to close all pending issues before the repo migration.
This logic is moved to the Migration Struct that should be included in the `frame_executive::Executive` when a migration needs to be passed. 

My understanding is that `on_runtime_upgrade` hooks are deprecated and the migration logic should move out of the pallet into these Migration structs so that they can be controlled and configured by runtime authors.

This is why I opened this PR in the first place, so that `pallet_xcm` adheres to this new conventions and integrate better with tools like try-runtime

Reply:
Yes, but what we're saying here in this comment is whether or not the existing runtimes that we care about is already on storage version 1.

Reply:
I don't think this should impact whether or not we merge this PR? 
The code here was moved to MigrateToV1.
If one runtime is not on v1 then it will run the upgrade by passing MigrateToV1 to frame_executive
if it's already on V1, then you just don't use MigrateToV1.
In both scenarios dry-running the migration with try-runtime will tell you if something is missing or superfluous.

Feel free to close to unblock the repo migration if I got that wrong